### PR TITLE
REACTOR-362 - Extend CZ-NEF to store Access Key Id Used and Access Key Id Granted

### PR DESCRIPTION
**Pull request by tiyberius**
### Description

- [ ] Existing GitHub Issue? Please provide #: 

### Acceptance checklist

- [ ] Feedback addressed
- [ ] Tests added or modified appropriately
- [ ] All tests passed

### CODE REVIEW

**Comment by tiyberius** [comment]
Ah, @imcgaunn I'm glad you asked about `id_granted` vs `id_used`. That is a very important distinction, so I'd like to find the correct terminology. The Access Key Id `granted` is the Access Key Id given to a user or an account by AWS when it authenticates successfully. And when that user or account USES that Access Key Id later, that is the `id_used`

File: reactor/aws/tools.py [hunk 1]
```diff
@@ -589,3 +592,59 @@ def get_reactor_functions(namespace):
...
+
+
+def get_access_key_id_granted_in_event(event):
+    """
+    >>> get_access_key_id_granted_in_event({'responseElements': {'credentials': {'accessKeyId': 'flargy'}}})
+    'flargy'
+    >>> get_access_key_id_granted_in_event(None)
+    Traceback (most recent call last):
+    ...
+    TypeError: 'NoneType' object is not subscriptable
```
Comment (killthrush) [inline reactor/aws/tools.py]: this can also raise key errors.  Add a test case?

File: reactor/aws/tools.py [hunk 2]
```diff
@@ -589,3 +592,59 @@ def get_reactor_functions(namespace):
...
+    access_key_id = event['responseElements']['credentials']['accessKeyId']
+    return access_key_id
+
+
+def get_access_key_id_granted_in_event_if_exists(event):
+    """
+    >>> get_access_key_id_granted_in_event_if_exists({'responseElements': {'credentials': {'accessKeyId': 'flargy'}}})
+    'flargy'
+    >>> get_access_key_id_granted_in_event_if_exists({})
+    >>> get_access_key_id_granted_in_event_if_exists(None)
```
Comment (killthrush) [inline reactor/aws/tools.py]: So if there's no value on the line below the test case/example that means 'this function returns None'?

File: reactor/features/simplequery/adapters.py [hunk 3]
```diff
@@ -53,6 +53,8 @@
     externalId: STRING >,
   event_resources ARRAY <
     STRING >,
+  access_key_id_used STRING,
+  access_key_id_granted STRING,
```
Comment (killthrush) [inline reactor/features/simplequery/adapters.py]: Do we need to add these to CZ-NEF and to aurora?

File: reactor/aws/tools.py (hunk 2 above)
Comment (tiyberius) [reply reactor/aws/tools.py]: That is correct. A bit confusing, but that's what happens in the Python interpreter, so it's what happens here too

File: reactor/features/simplequery/adapters.py (hunk 3 above)
Comment (tiyberius) [reply reactor/features/simplequery/adapters.py]: Yes! We totally need to. What do you think about the column names ... do they make sense?

File: reactor/aws/tools.py [hunk 4]
```diff
@@ -589,3 +592,59 @@ def get_reactor_functions(namespace):
...
+    >>> get_access_key_id_used_in_event_if_exists({'userIdentity': {'accessKeyId': 'flargy'}})
+    'flargy'
+    >>> get_access_key_id_used_in_event_if_exists({})
+    >>> get_access_key_id_used_in_event_if_exists(None)
+    """
+    access_key_id = None
+    try:
+        access_key_id = get_access_key_id_used_in_event(event)
+    except (KeyError, TypeError):
+        logger.debug(f'Unable to get {access_key_id} used in event')
```
[the remark this answers is not in the document]
Comment (tiyberius) [reply reactor/aws/tools.py]: done

File: reactor/aws/tools.py (hunk 1 above)
Comment (tiyberius) [reply reactor/aws/tools.py]: great catch, will be fixed in next update

**Pull request merged**